pkg/patterns/declarative/kustomize/enabled.go

```diff
@@ -0,0 +1,38 @@
+package kustomize
+
+import (
+	"context"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+func init() {
+	Kustomize = &EnabledKustomize{}
+}
+
+var _ Kustomizer = &EnabledKustomize{}
+
+// EnabledKustomize runs kustomize build. It requires kustomize/api v0.12.1 and above
+type EnabledKustomize struct{}
+
+// Run calls the kustomize/api library to run `kustomize build`.
+func (k *EnabledKustomize) Run(ctx context.Context, fs filesys.FileSystem, manifestPath string) ([]byte, error) {
```

Comment:
FYI you can have the same function with a build tag and without the build tag.

So you could declare `RunKustomize` in both.

One would have the build-tag `//go:build without_kustomize`, the other would have the build tag `//go:build !without_kustomize`.

Not a blocker though - and the refactoring is nice - but may be simpler in future.

Reply:
Thank you for the suggestion! I'm trying to use function/struct name to make the code self-explained. But I agree the `EnabledKustomize` and `SkipKustomize` structs may give duplicate info since the go file name ("disabled" "enabled) and the build tags have already explained the rationale. Updated!